### ISSUE TITLE
Several "orphaned" pages in Lore format documentation

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4239]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)| @khorn reported|
|-|-|
|Trac ID|trac#4239|
|Type|defect|
|Created|2010-01-28 20:27:13Z|
|Branch|https://github.com/twisted/twisted/tree/orphaned-pages-4239|

The following pages exist in the Twisted documentation, but nothing links to them.  They are not listed in any index pages, and do not appear to be linked from any other pages either.

```
/web/howto/formindepth.xhtml
/core/howto/pclients.xhtml
/core/howto/upgrading.xhtml
/pair/examples/index.xhtml
```

<details><summary>Searchable metadata</summary>

```
trac-id__4239 4239
type__defect defect
reporter__khorn khorn
priority__normal normal
milestone__ 
branch__branches_orphaned_pages_4239 branches/orphaned-pages-4239
branch_author__khorn khorn
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1264710433000000 1264710433000000
changetime__1269290455000000 1269290455000000
version__None None
owner__ 
cc__thijs cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_1"></a>|@khorn commented|
|-|-|

`/web/howto/formindepth.xhtml` - Just contains "XXX: to be written", so it probably should be.  It would be helpful to know what this document was intended to cover...

`/core/howto/pclients.xhtml` - Has content, but just isn't linked to for some unknown reason.  Also, perhaps this should be renamed "pb-clients" or something similar, to follow the pattern of the other Perspective Broker docs?

`/core/howto/upgrading.xhtml` - Appears to be all about "versioned" PB applications. Perhaps it also needs to be renamed?

`/pair/examples/index.xhtml` - This just looks like a missing link.  Shouldbe an easy fix.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

Can you branch for this so we can test with the doc build slave?

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_3"></a>|@khorn commented|
|-|-|

Apparently mkbranch from Combinator doesn't work on Windows at all, and that's what I have working right now.  So I can't make branches from Combinator.  Either someone else can create the branch, or I can create one using SVN, but I'm not clear on whether mkbranch just creates the branch normally or if it does something special with respect to Combinator.

Please advise...

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

Making a branch without Combinator just involves the usual svn stuff.  The only thing to pay attention to is that you should copy trunk in a _different_ revision from applying any changes to it.

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_5"></a>|@khorn commented|
|-|-|

(In [28168]) Branching to 'orphaned-pages-4239'

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_6"></a>|@khorn removed owner|
|-|-|

proposed changes in [28717]

sending to review...

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_8"></a>|@jesstess set owner to @khorn|
|-|-|

* Nothing in twisted/pair/* has a copyright above 2004, and running the pairudp.py example given a link in the pair examples index in this branches gets me ```ImportError: No module named eunuchs.tuntap```. If pairudp.py is going to be linked, it should probably have a shebang line and a copyright like other examples. Whether or not this example should even still exist probably needs perspective from someone who knows if Pair is theoretically maintained still (aka not me).
* "Miscellaneous" is spelled incorrectly on the pair index.xhtml
* updated.xhtml and pb-client.xhtml talk about things that aren't deprecated and have even been updated recently in some cases, so it seems plausible that they really should be linked from the howto. Both have trailing whitespace. I didn't do a full documentation check on them because that would blow up changes on this ticket. If you're interested, file another another one for markup issues in these files and consider linking good enough.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_9"></a>|@jesstess commented|
|-|-|

Formatting fail:

 * Nothing in twisted/pair/* has a copyright above 2004, and running the pairudp.py example given a link in the pair examples index in this branches gets me ImportError: No module named eunuchs.tuntap. If pairudp.py is going to be linked, it should probably have a shebang line and a copyright like other examples. Whether or not this example should even still exist probably needs perspective from someone who knows if Pair is theoretically maintained still (aka not me).
 * "Miscellaneous" is spelled incorrectly on the pair index.xhtml
 * updated.xhtml and pb-client.xhtml talk about things that aren't deprecated and have even been updated recently in some cases, so it seems plausible that they really should be linked from the howto. Both have trailing whitespace. I didn't do a full documentation check on them because that would blow up changes on this ticket. If you're interested, file another another one for markup issues in these files and consider linking good enough.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_10"></a>|@jesstess commented|
|-|-|

Also formindepth.xhtml from the ticket summary isn't currently addressed (deleted?) in the branch.

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_11"></a>|@khorn commented|
|-|-|

actually I left formindepth.xhtml alone...since it's effectively empty and I had no idea what it was intended to be.

I meant to note that here in the ticket, but apparently forgot.

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_12"></a>|@khorn commented|
|-|-|

(In [28753]) Address review comments

 * remove training whitespace
 * spelling fix

Refs: [#4239](https://github.com/twisted/twisted/issues/4239)

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)|@khorn removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_14"></a>|@jesstess set owner to @khorn|
|-|-|

Thanks for pushing through these documentation improvements, khorn.
 * If pairudp.py is going to stay linked, it needs a shebang line and a copyright like other examples, and a ticket to get it working.
 * I think formindepth.xhtml should just be deleted. I don't see any tickets related to completing it.

I see now that [TwistedProjects](https://twistedmatrix.com/trac/wiki/TwistedProjects) lists Twisted Pair as unmaintained. We need to get someone to admit the true state of Twisted Pair on IRC or the mailing list so we can work on removing it if that's what needs to happen. 

Other than that, looks good to merge!

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_15"></a>|@khorn commented|
|-|-|

(In [28761]) Address further review comments

 * remove /doc/web/howto/formindepth.xhtml
 * add copyright header to /doc/pair/examples/pairudp.py

Refs: [#4239](https://github.com/twisted/twisted/issues/4239)

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_16"></a>|@khorn set status to `closed`|
|-|-|

(In [28766]) Fix links for orphaned pages in docs

 * remove doc/web/howto/formindepth.xhtml
 * rename doc/core/howto/pclients.xhtml (to pb-clients.xhtml)
 * add copyright header and shebang line to pair/examples/pairudp.py
 * added some links to point to orphaned pages

Author: khorn
Reviewer: jesstess
Fixes: [#4239](https://github.com/twisted/twisted/issues/4239)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
